src/main/kotlin/kotcity/data/CityFileAdapter.kt

```diff
@@ -216,51 +216,56 @@ object CityFileAdapter: Debuggable {
 
                 if (fromBuildings.count() == 0) {
                     println("Error during contact loading! Cannot find source building at $from")
-                } else if (toBuildings.count() == 0) {
+                    return@forEach
+                }
+                if (toBuildings.count() == 0) {
                     println("Error during contact loading! Cannot find source building at $to")
-                } else {
-                    val fromBuilding = fromBuildings.first()
-                    val toBuilding = toBuildings.first()
-                    val tradeable = Tradeable.valueOf(contractObj["tradeable"].asString)
-                    val quantity = contractObj["quantity"].asInt
-                    // val newContract = Contract(fromBuilding, toBuilding, tradeable, quantity)
-
-                    val path = pathfinder.tripTo(listOf(from), listOf(to))
-                    if (path != null) {
-                        toBuilding.building.createContract(CityTradeEntity(fromBuilding.coordinate, fromBuilding.building), tradeable, quantity, path)
-                    } else {
-                        println("Error during contract loading! Can't find path!")
-                    }
+                    return@forEach
                 }
 
+                val fromBuilding = fromBuildings.first()
+                val toBuilding = toBuildings.first()
+                val tradeable = Tradeable.valueOf(contractObj["tradeable"].asString)
+                val quantity = contractObj["quantity"].asInt
 
+                val path = pathfinder.tripTo(listOf(from), listOf(to))
 
+                if (path == null) {
+                    println("Error during contract loading! Can't find path!")
+                    return@forEach
```

Comment:
Why isn't there a space after return?

Reply:
thats kotlin syntax. it's a named return and sometimes necessary to make it clear what scope should actually be returned from.